Device login failed with high probability

### Environment:
```
Bits: Dev1077
OS: Windows10/Mac/Ubuntu
Eclipse: 2019-03
```
### Repro Steps:
1. Open `Azure Sign In` dialog
2. Select `Device login` or `Service Principal`-> `New `login method
3. Click `Copy&Open` button in `Azure Device Login` dialog
4. Paste the code in the website
5. Type your account info

### Result:
The login is successful on the website but failed in Eclipse
![image](https://user-images.githubusercontent.com/22470312/55857009-97b9b000-5b9e-11e9-812d-1a4f7b662b89.png)
![image](https://user-images.githubusercontent.com/22470312/55857013-9ab4a080-5b9e-11e9-980b-5a45a7c231ee.png)

### Log
```
java.util.concurrent.ExecutionException: java.net.SocketTimeoutException: Read timed out
	at java.util.concurrent.FutureTask.report(Unknown Source)
	at java.util.concurrent.FutureTask.get(Unknown Source)
	at com.microsoft.azuretools.core.ui.DeviceLoginWindow$DeviceLoginDialog.pullAuthenticationResult(DeviceLoginWindow.java:181)
	at com.microsoft.azuretools.core.ui.DeviceLoginWindow$DeviceLoginDialog.lambda$0(DeviceLoginWindow.java:102)
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(Unknown Source)
	at java.net.SocketInputStream.read(Unknown Source)
	at java.net.SocketInputStream.read(Unknown Source)
	at sun.security.ssl.InputRecord.readFully(Unknown Source)
	at sun.security.ssl.InputRecord.read(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.readRecord(Unknown Source)
	at sun.security.ssl.SSLSocketImpl.readDataRecord(Unknown Source)
	at sun.security.ssl.AppInputStream.read(Unknown Source)
	at java.io.BufferedInputStream.fill(Unknown Source)
	at java.io.BufferedInputStream.read1(Unknown Source)
	at java.io.BufferedInputStream.read(Unknown Source)
	at sun.net.www.http.HttpClient.parseHTTPHeader(Unknown Source)
	at sun.net.www.http.HttpClient.parseHTTP(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
	at java.net.HttpURLConnection.getResponseCode(Unknown Source)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(Unknown Source)
	at com.microsoft.aad.adal4j.AdalOAuthRequest.processAndReadResponse(AdalOAuthRequest.java:155)
	at com.microsoft.aad.adal4j.AdalOAuthRequest.send(AdalOAuthRequest.java:84)
	at com.microsoft.aad.adal4j.AdalTokenRequest.executeOAuthRequestAndProcessResponse(AdalTokenRequest.java:86)
	at com.microsoft.aad.adal4j.AuthenticationContext.acquireTokenCommon(AuthenticationContext.java:928)
	at com.microsoft.aad.adal4j.AcquireTokenCallable.execute(AcquireTokenCallable.java:70)
	at com.microsoft.aad.adal4j.AcquireTokenCallable.execute(AcquireTokenCallable.java:38)
	at com.microsoft.aad.adal4j.AdalCallable.call(AdalCallable.java:47)
	... 4 more

```

--- a/PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/DeviceLoginWindow.java
+++ b/PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/ui/DeviceLoginWindow.java
@@ -192,7 +192,8 @@ public class DeviceLoginWindow implements IDeviceLoginUI {
                     if (e.getCause() instanceof AuthenticationException &&
                         ((AuthenticationException) e.getCause()).getErrorCode()
                             == AdalErrorCode.AUTHORIZATION_PENDING) {
-                        // swallow the pending exception
+                        // reset the retryCount to zero, will quit only for 3 consecutive fail
+                        checkTime = 0;
                     } else {
                         checkTime++;
                         LOG.log(new Status(IStatus.ERROR, Activator.PLUGIN_ID, "DeviceLoginWindow", e));

